securedrop_client/logic.py

```diff
@@ -439,6 +470,7 @@ def on_update_star_success(self, result) -> None:
         """
         After we star a source, we should sync the API such that the local database is updated.
         """
+        self.set_status('')  # remove any permanent error status message
```

Comment:
I see why you're doing this: the next job that _might_ succeed isn't necessarily a `MetadataSyncJob`, hence needing `self.set_status` sprinkled in our success callbacks. We should put this somewhere generic, some options:

1. we reach into the gui object from the queue - technically "works" but this transitive navigation is poor design
2. we emit a separate signal indicating that the error status should be cleared - I guess that works too but I don't love it - we already have a success signal getting emitted and it would unfortunate to complicate the design of the queue for this purpose
3. we create a decorator `@success_callback` and put any per-success logic in there along with our explanation of why it's needed, and then decorate any success callbacks with it. It's the same size diff, but it's a bit more extensible. We could also do the same for the failure callback to ensure that all failures have a log line (since most of our failure callbacks have a debug line to log a failure, we could put that in a generic place too to ensure that we're logging these failures). 

anyway typing this out I'm liking 3 but also feeling like this should be a followup - what do you think @creviera ?

Reply:
Yes, I agree with #3 and it being a followup issue

Reply:
I agree with #3 and captured this in a followup issue: https://github.com/freedomofpress/securedrop-client/issues/651